tests/test_core.py

```diff
@@ -1118,6 +1118,40 @@ class MyClass(NSObject):
             class MyClass(NSObject):  # noqa: F811
                 pass
 
+    def test_class_auto_rename(self):
+        """Test automatic renaming system of ObjCClass."""
+
+        ObjCClass.auto_rename = True
+
+        for _ in range(2):
+
+            class TestClass2(NSObject):
+                pass
+
+        ObjCClass.auto_rename = False
+
+        for _ in range(2):
+
+            class TestClass(NSObject, auto_rename=True):
+                pass
```

Comment:
This test exercises the fact that the code doesn't raise an exception, but doesn't verify anything about the operation of the code. e.g.:

* The ObjC name of the resulting class is unique, and follows the expected pattern
* The renaming scheme skips names allocated manually that would collide (e.g., if I explicitly declare `TestClass_2`, that name won't be allocated)
* The TestClass instances generated by different renames don't compare as equal
* When a class name is reused, the new method implementations are also used
* The test isn't dependent on the specific order in which the test is invoked.